src/main/scala/encry/storage/levelDb/versionalLevelDB/VersionalLevelDB.scala

```diff
@@ -0,0 +1,407 @@
+    *
+    * @return
+    */
+  def getAll: List[(VersionalLevelDbKey, VersionalLevelDbValue)] = {
+    val readOptions = new ReadOptions()
+    readOptions.snapshot(db.getSnapshot)
+    val result = getCurrentElementsKeys
+      .foldLeft(List.empty[(VersionalLevelDbKey, VersionalLevelDbValue)]) {
+        case (acc, nextKey) =>
+          (nextKey -> get(nextKey).get) :: acc
+      }
+    readOptions.snapshot().close()
+    result
+  }
+
+  /**
+    * Remove all insertions of version after rollbackPoint
+    *
+    * @return
+    */
+  @tailrec
+  private def rollbackResolver(versionsToResolve: List[LevelDBVersion]): Unit = {
+    if (versionsToResolve.nonEmpty) {
+      val versionToResolve = versionsToResolve.head
+      val readOptions = new ReadOptions()
+      readOptions.snapshot(db.getSnapshot)
+      val writeBatch = db.createWriteBatch()
+      val insertions = splitValue2elems(DEFAULT_USER_KEY_SIZE, db.get(versionKey(versionToResolve), readOptions))
+      val deletions = splitValue2elems(DEFAULT_USER_KEY_SIZE, db.get(versionDeletionsKey(versionToResolve), readOptions))
+      insertions.foreach { elemKey =>
+        writeBatch.delete(accessableElementKeyForVersion(versionToResolve, VersionalLevelDbKey @@ elemKey))
+        val accessMap = db.get(userKey(VersionalLevelDbKey @@ elemKey), readOptions)
+        //only one version contains elem with this key, so remove it
+        if (accessMap.length == settings.versionKeySize + 1)
+          writeBatch.delete(userKey(VersionalLevelDbKey @@ elemKey))
+        else {
+          val versions =
+            splitValue2elems(settings.versionKeySize, accessMap.tail).map(ver => new ByteArrayWrapper(ver))
+              .dropWhile(_ != new ByteArrayWrapper(versionToResolve))
+          val vertoput = versions.tail
+            .foldLeft(Array.emptyByteArray) {
+              case (acc, elem) => acc ++ elem.data
+            }
+          writeBatch.put(userKey(VersionalLevelDbKey @@ elemKey), ACCESSIBLE_KEY_PREFIX +: vertoput)
+        }
+      }
+      deletions.foreach { elemKey =>
+        val accessMap = db.get(userKey(VersionalLevelDbKey @@ elemKey), readOptions)
+        writeBatch.put(userKey(VersionalLevelDbKey @@ elemKey), ACCESSIBLE_KEY_PREFIX +: accessMap.tail)
+      }
+      db.write(writeBatch)
+      writeBatch.close()
+      readOptions.snapshot().close()
+      rollbackResolver(versionsToResolve.tail)
+    }
+  }
+
+  /**
+    * Remove unused keys
+    * if version elems keys is "resolved". Set key "SERVICE_PREFIX :: versrion ++ RESOLVED" prefix to true (1:Byte)
+    *
+    * @return
+    */
+  //todo: refactor
+  def deleteResolver(versionToResolve: LevelDBVersion): Unit = {
+    val batch = db.createWriteBatch()
+    val readOptions = new ReadOptions()
+    readOptions.snapshot(db.getSnapshot)
+    val wrappedVer: ByteArrayWrapper = new ByteArrayWrapper(versionToResolve)
+    val currentVersions: List[ByteArrayWrapper] = versionsList(readOptions).map(ByteArrayWrapper.apply)
```

Comment:
It's better to make it a Set, because we only use it to test `contains` predicate